sw/device/silicon_creator/lib/cert/asn1.h

```diff
@@ -25,8 +25,17 @@ typedef struct asn1_state {
   size_t size;
   // Current offset in the output.
   size_t offset;
+  // Last error.
+  rom_error_t error;
 } asn1_state_t;
 
+/**
+ * Clear the ASN1 builder internal error status.
+ *
+ * @param[out] state Pointer to a user-allocated state to be cleared.
+ */
+void asn1_clear_error(asn1_state_t *state);
+
 /**
```

Comment:
It think it's very important to write somewhere (maybe everywhere?) that every operation can update the `error` variable in the state.

Reply:
Sure things, added in all function comments.